aiidalab_qe/steps.py

```diff
@@ -433,10 +485,24 @@ def _auto_select_code(self, change):
         if change["new"] and not change["old"]:
             parameters = self.get_input_parameters()
             self.codes_selector.pw.refresh()
+            self.codes_selector.dos.refresh()
+            self.codes_selector.projwfc.refresh()
             try:
                 self.codes_selector.pw.selected_code = load_code(parameters["pw_code"])
             except NotExistent:
                 pass
+            try:
+                self.codes_selector.dos.selected_code = load_code(
+                    parameters["dos_code"]
+                )
+            except NotExistent:
+                pass
+            try:
+                self.codes_selector.projwfc.selected_code = load_code(
+                    parameters["projwfc_code"]
+                )
+            except NotExistent:
+                pass
```

Comment:
I would suggest that we move the utility `_load_code()` function from line 582/561 into the module namespace and then use it here as well for improved readability.

```suggestion
        self.codes_selector.pw.selected_code = _load_code(parameters["pw_code"])
        self.codes_selector.dos.selected_code = _load_code(parameters["dos_code"])
        self.codes_selector.projwfc.selected_code = _load_code(
            parameters["projwfc_code"]
        )
```

The function itself could be expressed a bit more succinctly like this:

```python
from contextlib import suppress

def _load_code(code):
    with suppress(NotExistent, ValueError):
        return load_code(None)
```

Reply:
Looks very cool! changed.